pkg/virt-controller/services/template_test.go

```diff
@@ -5100,6 +5101,175 @@ var _ = Describe("Template", func() {
 			Expect(netBindingPluginMemoryOverheadCalculator.calculatedMemoryOverhead).To(BeTrue())
 		})
 	})
+
+	Context("Custom annotations Generation", func() {
+		const (
+			generator1Key1   = "generator1Key1"
+			generator1Value1 = "generator1Value1"
+			generator1Key2   = "generator1Key2"
+			generator1Value2 = "generator1Value2"
+
+			generator2Key1   = "generator2Key1"
+			generator2Value1 = "generator2Value1"
+			generator2Key2   = "generator2Key2"
+			generator2Value2 = "generator2Value2"
+		)
+
+		var (
+			vmi *v1.VirtualMachineInstance
+
+			generator1Annotations map[string]string
+			generator2Annotations map[string]string
+
+			generator1Err = errors.New("generator1 failed")
+			generator2Err = errors.New("generator2 failed")
+		)
+
+		BeforeEach(func() {
+			vmi = libvmi.New(libvmi.WithNamespace("default"))
+
+			generator1Annotations = map[string]string{
+				generator1Key1: generator1Value1,
+				generator1Key2: generator1Value2,
+			}
+
+			generator2Annotations = map[string]string{
+				generator2Key1: generator2Value1,
+				generator2Key2: generator2Value2,
+			}
+		})
+
+		It("Should call all registered annotation generators", func() {
+			generator1 := stubAnnotationsGenerator{
+				annotations: generator1Annotations,
+			}
+			generator2 := stubAnnotationsGenerator{
+				annotations: generator2Annotations,
+			}
+
+			svc = NewTemplateService("kubevirt/virt-launcher",
+				240,
+				"/var/run/kubevirt",
+				"/var/lib/kubevirt",
+				"/var/run/kubevirt-ephemeral-disks",
+				"/var/run/kubevirt/container-disks",
+				v1.HotplugDiskDir,
+				"pull-secret-1",
+				pvcCache,
+				virtClient,
+				config,
+				qemuGid,
+				"kubevirt/vmexport",
+				resourceQuotaStore,
+				namespaceStore,
+				WithAnnotationsGenerators(generator1, generator2),
+			)
+
+			pod, err := svc.RenderLaunchManifest(vmi)
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(pod.Annotations).To(HaveKeyWithValue(generator1Key1, generator1Value1))
+			Expect(pod.Annotations).To(HaveKeyWithValue(generator1Key1, generator1Value1))
+			Expect(pod.Annotations).To(HaveKeyWithValue(generator2Key1, generator2Value1))
+			Expect(pod.Annotations).To(HaveKeyWithValue(generator2Key2, generator2Value2))
+		})
+
+		DescribeTable("Should fail when the first generator failure is encountered", func(generator1, generator2 stubAnnotationsGenerator, expectedErr error) {
```

Comment:
Since `WithAnnotationsGenerators` is a varadiac function that takes in the interface type:
```go
func WithAnnotationsGenerators(generators ...annotationsGenerator) templateServiceOption
```

It is not possible to pass it `[]stubAnnotationsGenerator` directly:
```go
WithAnnotationsGenerators(generators...)
```

It seems that we would need to convert the `[]stubAnnotationsGenerator` to`[]annotationsGenerator` first - which will just add more code and wouldn't provide any additional value.